Update BMC styles

rmzelle opened the pull request:
Alternative PR to #1534.

Accompanied by https://github.com/citation-style-language/utilities/commit/b5297f086c780e7210f4c936e4e611eb04ae4a47 in the utilities repo and https://docs.google.com/spreadsheets/d/1ryYTyRg9c5m8ecg4ak0SFM94-jq0K6pD7XV5e0gc9Pg/edit?usp=sharing.

Not really sure which approach is better (fixing up the Springer data or updating the BMC data).

zuphilip commented:
Thank you for the ongoing work on these styles. I will be away this week and might not have time to look at the PR soon. However, I guess that might not be necessary at all.

rmzelle commented:
Well, it would be good to have a second pair of eyes on this.

In short, I took the BMC journal list, removed the journals with no listed citation style (all discontinued), and all styles that have "Springer" as their publisher, and removed all duplicates. I then used the following translation table:

```
VANCOUVER	springer-vancouver-brackets	numeric
CHEMISTRY	springer-basic-brackets-no-et-al	numeric
BIOMEDCENTRAL	springer-vancouver-brackets	numeric
BASIC_SPRINGER	springer-basic-author-date	author-date
```

(would be good to double check this)

zuphilip commented on dependent/acta-neuropathologica-communications.csl:
```diff
@@ -6,9 +6,9 @@
+    <link href="http://www.zotero.org/styles/springer-basic-author-date" rel="independent-parent"/>
     <link href="http://www.actaneurocomms.org" rel="documentation"/>
-    <category citation-format="numeric"/>
+    <category citation-format="author-date"/>
```
This seems to be correct currently and the replacement looks wrong here, see http://www.actaneurocomms.org/authors/instructions/research#formatting-references and e.g. http://www.actaneurocomms.org/content/3/1/32

zuphilip commented on dependent/amb-express.csl:
```diff
@@ -1,18 +1,16 @@
 <?xml version="1.0" encoding="utf-8"?>
 <style xmlns="http://purl.org/net/xbiblio/csl" version="1.0" default-locale="en-US">
-  <!-- Generated with https://github.com/citation-style-language/utilities/tree/master/generate_dependent_styles/data/bmc -->
+  <!-- Generated with https://github.com/citation-style-language/utilities/tree/master/generate_dependent_styles/data/springer -->
```
Shouldn't the generated info `generate_dependent_styles/data/bmc` always change? Or why does it change here, but not for the cases above?

rmzelle replied:
This is a Springer published journal, so I excluded it from the BMC dataset, relying instead on the Springer dataset.

zuphilip commented on dependent/annals-of-intensive-care.csl:
```diff
@@ -1,18 +1,16 @@
     <link href="http://www.zotero.org/styles/annals-of-intensive-care" rel="self"/>
-    <link href="http://www.zotero.org/styles/biomed-central" rel="independent-parent"/>
-    <link href="http://www.annalsofintensivecare.com" rel="documentation"/>
+    <link href="http://www.zotero.org/styles/springer-basic-brackets" rel="independent-parent"/>
```
That should be `springer-vancouver-brackets`.

zuphilip commented on dependent/bundesgesundheitsblatt-gesundheitsforschung-gesundheitsschutz.csl:
```diff
@@ -1,15 +1,14 @@
     <link href="http://www.zotero.org/styles/bundesgesundheitsblatt-gesundheitsforschung-gesundheitsschutz" rel="self"/>
-    <link href="http://www.zotero.org/styles/springer-fachzeitschriften-medizin-psychologie" rel="independent-parent"/>
-    <link href="http://www.springer.com/medicine/journal/103?detailsPage=pltci_1426209" rel="documentation"/>
+    <link href="http://www.zotero.org/styles/springer-basic-brackets" rel="independent-parent"/>
```
We should check again, because it has its own Endnote style.

zuphilip replied:
Yes, that is fine, see https://github.com/citation-style-language/styles/issues/1583#issuecomment-110794540

zuphilip commented on dependent/carbon-balance-and-management.csl:
```diff
@@ -1,18 +1,16 @@
     <link href="http://www.zotero.org/styles/carbon-balance-and-management" rel="self"/>
-    <link href="http://www.zotero.org/styles/biomed-central" rel="independent-parent"/>
-    <link href="http://www.cbmjournal.com" rel="documentation"/>
+    <link href="http://www.zotero.org/styles/springer-basic-brackets" rel="independent-parent"/>
```
Should be vancouver

zuphilip commented on dependent/clinical-and-translational-medicine.csl:
```diff
@@ -1,18 +1,16 @@
     <link href="http://www.zotero.org/styles/clinical-and-translational-medicine" rel="self"/>
-    <link href="http://www.zotero.org/styles/biomed-central" rel="independent-parent"/>
-    <link href="http://www.clintransmed.com" rel="documentation"/>
+    <link href="http://www.zotero.org/styles/springer-basic-brackets" rel="independent-parent"/>
```
Should be vancouver

zuphilip commented:
I checked the styles from A to C so far...

zuphilip commented:
@rmzelle Actually, I have forgotten about this PR. What is the status here? (Note: I would like to ask Springer soon for a new list of their used styles.)

rmzelle commented:
BMC never bothered to directly respond to my repeated inquiries, but Matias Piipari from http://www.manuscriptsapp.com/ (and previously from Papers) had some contacts at BMC which helped to escalate the issue, and it looks like BMC finally updated their journal metadata file in the last month: http://old.biomedcentral.com/journals/biomedcentraljournallist.txt (previously it had a lot of double entries per journal with differing citation styles)

I'll take a look soon.

rmzelle commented:
Looks like all journals in the new list have "VANCOUVER" as the citation style, by the way.

zuphilip commented:
Yes, I have also received an answer in May from my contact at Springer quoting someone at BMC:
<blockquote>All journals that were previously BioMed Central (BMC) Reference Style or Springer Open Reference Style will have now changed to Vancouver. Any of our journals that were not previously using one of these reference styles will have stayed the same. No journals have changed from BMC to Springer style.</blockquote>

rmzelle commented:
It might be worthwhile to ask for new Springer data. Maybe we don't need the BMC data at all.

zuphilip commented:
Okay, I just checked the new list and it looks like everything (minus few) styles are already covered currently by the springer list: https://docs.google.com/spreadsheets/d/102VO4EkV0QBofUE45GG1U517s4cArBHsX02uy8C7pEs/edit?usp=sharing

Thus, it may be a good idea to ask for the springer list and check if we may drop the BMC completely.

adam3smith commented:
superseded by https://github.com/citation-style-language/styles/pull/1885
